server/api.js

```diff
@@ -2,100 +2,416 @@ import { Router } from "express";
+						status: 404,
+						msg: "This user doesn't have a goal with this id",
+					});
+				}
+			})
+			.catch((err) => console.error(err.message));
+	}
 );
-res.json(newUser.rows[0]);
-} catch (err) {
-console.error(err.message);
-}
+
+// GET all feedbacks form specific user id
+router.get("/users/:user_id/feedbacks", async (req, res) => {
+	const { user_id } = req.params;
+	try {
+		const feedback_req = await pool.query(
+			`SELECT graduate.first_name, graduate.id, p.description, f.description, f.id, mentors.first_nameFROM feedbacks as f 
+				INNER JOIN plans  as p ON p.id = f.plan_id INNER JOIN users as graduate ON graduate.id = p.user_id
+				INNER JOIN users as mentors ON f.user_id= mentors.id where mentors.id = $1
+					ORDER BY f.create_date`,
+			[user_id]
+		);
+		res.json(feedback_req.rows);
+	} catch (err) {
+		res.status(500).json("server error");
+	}
 });
 
-router.get("/users/:id", async (req, res) => {
+// POST feedback form specific user id (mentor)
+router.post("/users/:user_id/feedbacks", async (req, res) => {
+	const { id } = req.body;
+	const { user_id } = req.params;
+
 	try {
-		console.log(req.body);
-		const { id } = req.params;
-		const users = await pool.query(
-			"SELECT * FROM users WHERE id = $1",
-			[id]
+		await pool.query(
+			"INSERT INTO feedbacks(plan_id, user_id) VALUES ($1,$2)",
+			[id, user_id]
 		);
-		res.json(users.rows[0]);
+
+		res.json("feedback send successfully");
 	} catch (err) {
-		console.error(err.message);
+		res.status(500).json(err, "server error");
 	}
 });
 
-router.get("/users/:id/plans", async (req, res) => {
+// UPDATE feedback form specific user id (mentor) and user plan_id
+router.put("/users/:user_id/:plan_id/feedbacks", async (req, res) => {
+	const { description } = req.body;
+	const { plan_id, user_id } = req.params;
+
 	try {
-		console.log(req.body);
-		const { id } = req.params;
-		const plans = await pool.query(
-			"SELECT * FROM plans WHERE user_id = $1", [id]);
-		res.json(plans.rows[0]);
+		await pool.query(
+			"UPDATE feedbacks SET description=$1 WHERE plan_id=$2 AND users_id=$3",
+			[description, plan_id, user_id]
+		);
+
+		res.json("feedback send successfully");
 	} catch (err) {
-		console.error(err.message);
+		res.status(500).send(err, "server error");
 	}
 });
 
-router.get("/users/:id/:plan_id", async (req, res) => {
+// GET all feedbacks form specific user id
+router.get("/users/:user_id/feedbacks", async (req, res) => {
 	try {
-		console.log(req.body);
-		const { id, plan_id } = req.params;
-		const goals = await pool.query("SELECT * FROM plans WHERE user_id = $1 AND plan_id = $2", [id, plan_id]);
-		res.json(goals.rows[0]);
+		const feedbacks = await pool.query("",[req.params.user_id]);
+		res.json(feedbacks.rows);
 	} catch (err) {
-		console.error(err.message);
+		res.status(500).send(err, "server error");
 	}
 });
 
+// PUT a feedback from specific plan id
+router.put("/user/:user_id/:plan_id/feedbacks", async (req, res) => {
```

Comment:
This is inconsistent with how you used params before.
Here `user_id` refers to the mentor leaving the feedback (or, generally, the _feedback author_), but in all other endpoints it was referring to the plan owner. This problem will be fixed after adding authentication.

But before we reach that point I suggest using `/plans/:plan_id/feedbacks` as the route and passing the user_id in the request body.